app/forms/search_form.rb

```diff
@@ -12,13 +12,17 @@ class SearchForm
   # TODO: Full text search should be considered for keyword search.
   # rubocop:disable Metrics/AbcSize
   def search
-    return Developer.none if email.present? && prog_lang.present? && language_code.present?
+
+    return Developer.none if email.nil? && language_code.nil? && prog_lang.nil?
 
     query = Developer.all
     query = query.where('email like :search_text', search_text: "%#{sanitize_sql_like(email)}%") if email.present?
-    query = query.joins(:programming_languages).where(programming_languages: {name: prog_lang}) if prog_lang.present?
-    query = query.joins(:languages).where(languages: {code: language_code}) if language_code.present?
+    query = query.where('name like :search_text', search_text: "%#{sanitize_sql_like(prog_lang)}%") if prog_lang.present?
+    query = query.where('code like :search_text', search_text: "%#{sanitize_sql_like(language_code)}%") if language_code.present?
+    query = query.joins(:programming_languages) if prog_lang.present?
+    query = query.joins(:languages) if language_code.present?
```

Comment:
The condition, `if language_code.present?`, duplicates line 21.
The same thing for line 20 and 22 regarding `if prog_lang.present?`.
There's no meaning to separate them.